src/app/clusters/level-control/level-control.cpp

```diff
@@ -171,9 +171,16 @@ class DefaultLevelControlSceneHandler : public scenes::DefaultSceneHandlerImpl
         uint8_t maxLevel;
         VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == Attributes::MaxLevel::Get(endpoint, &maxLevel), CHIP_ERROR_READ_FAILED);
 
-        pairs[0].attributeID    = Attributes::CurrentLevel::Id;
-        pairs[0].attributeValue = (!level.IsNull()) ? level.Value() : maxLevel + 1;
-        size_t attributeCount   = 1;
+        pairs[0].attributeID = Attributes::CurrentLevel::Id;
+        if (!level.IsNull())
+        {
+            pairs[0].attributeValue = level.Value();
+        }
+        else
+        {
+            chip::app::NumericAttributeTraits<uint32_t>::SetNull(pairs[0].attributeValue);
```

Comment:
But ApplyScene was not fixed, so this is broken....  How was this tested?

Reply:
Currently, If null value (255) for a uint8_t is stored, this will cause the ApplyScene to fail as we are asking to move to a value outside of the boundaries.

This will return failure to the Apply Scene. Is the expected behavior here to silently fail instead of explicitly letting the user know the scene wasn't applied?

Reply:
I would think the expected behavior is to move to the values that were stored in the scene, including a null level if that's what was stored.

Reply:
I see, so we do not want that recall scene to end up in a failure.

How do we expect the level to behave to a null level? Value capped at maximum or lowest possible value, remaining at current level?

So basically, we ignore the level command silently, we ignore the whole scene returning an error (current implementation), or we move to an unexpected level.

Say we explicitly add a scene with a null value, we could say this behaves as expected, but what if null is added from a save scene. This means the user just saved the current scene while the level value was null. This probably means the user was not expecting the level to have an impact when restoring the scene.

By that reasoning, should we decide that silently ignoring the move to null seems like the expected behavior?

Reply:
> How do we expect the level to behave to a null level?

Well, how was it behaving when the scene was saved?

The questions you are asking are spec questions, not implementation questions, no?  If the spec does not define this, it needs to...

Reply:
The MoveToLevel Command states that the command SHALL have the Level data field, constrained between 0 to 254. We do not respect that, therefore this should result in an error.

However the RecallScene states that the status of RecallScene at step 3 (applying EFS) SHALL be SUCCESS, so my initial interpretation seems wrong, silent failure seems to be the way here.

Reply:
The MoveToLevel Command states that the command SHALL have the Level data field, constrained between 0 to 254. We do not respect that, therefore this should result in an error.

However the RecallScene states that the status of RecallScene at step 3 (applying EFS) SHALL be SUCCESS, so my initial interpretation seems wrong, silent failure seems to be the way here.

Reply:
Where does the spec say that recalling a scene involves a MoveToLevel command?  Seems like if that's the intent the spec needs to say it...

Reply:
> set the attributes and corresponding state of the cluster accordingly.
The "corresponding state of the cluster" bits.
Either way, setting the attributes to null without taking an action on the actual level would mean read level won't reflect the actual state of the cluster if no action is taken.

Now does it have to be the actual moveToLevel command, the spec doesn't precise it. So we could technically create a separate function that allows us to move the level to "null" but that is still not defined right?

Reply:
@lpbeliveau-silabs Indeed.